typescript/cli/src/commands/warp.ts

```diff
@@ -427,6 +448,236 @@ export const check: CommandModuleWithContext<{
+      alias: ['rebalancerConfigFile', 'rebalancerConfig', 'configFile'],
+    },
+    checkFrequency: {
+      type: 'number',
+      description: 'Frequency to check balances in ms (defaults: 30 seconds)',
+      demandOption: false,
+      default: 60000,
+    },
+    withMetrics: {
+      type: 'boolean',
+      description: 'Enable metrics (default: true)',
+      demandOption: false,
+      default: false,
+    },
+    monitorOnly: {
+      type: 'boolean',
+      description: 'Run in monitor only mode (default: false)',
+      demandOption: false,
+      default: false,
+    },
+    manual: {
+      type: 'boolean',
+      description:
+        'Trigger a rebalancer manual run (default: false, requires --origin, --destination, --amount)',
+      demandOption: false,
+      implies: ['origin', 'destination', 'amount'],
+    },
+    origin: {
+      type: 'string',
+      description: 'The origin chain for manual rebalance',
+      demandOption: false,
+      implies: 'manual',
+    },
+    destination: {
+      type: 'string',
+      description: 'The destination chain for manual rebalance',
+      demandOption: false,
+      implies: 'manual',
+    },
+    amount: {
+      type: 'number',
+      description:
+        'The amount to transfer from origin to destination on manual rebalance. Defined in token units (E.g 100 instead of 100000000 wei for USDC)',
+      demandOption: false,
+      implies: 'manual',
+    },
+  },
+  handler: async ({
+    context,
+    config,
+    checkFrequency,
+    withMetrics,
+    monitorOnly,
+    manual,
+    origin,
+    destination,
+    amount,
+  }) => {
+    let monitor: Monitor;
+    let strategy: IStrategy;
+    let rebalancer: IRebalancer | undefined;
+    let metrics: Metrics | undefined;
+    let rebalancerConfig: RebalancerConfig;
+
+    try {
+      // Load rebalancer config from disk
+      rebalancerConfig = RebalancerConfig.load(config, {
+        checkFrequency,
+        withMetrics,
+        monitorOnly,
+      });
+      logGreen('✅ Loaded rebalancer config');
+
+      // Instantiate the factory used to create the different rebalancer components
+      const contextFactory = await RebalancerContextFactory.create(
+        rebalancerConfig,
+        context,
+      );
+
+      if (manual) {
+        // These values will be enforced when manual is true given the 'implies' option in the builder.
+        // This will probably never fail, but allows the type to be inferred as not undefined.
+        assert(origin, '--origin is required');
+        assert(destination, '--destination is required');
+        assert(amount, '--amount is required');
+
+        warnYellow(
+          `Manual rebalance strategy selected. Origin: ${origin}, Destination: ${destination}, Amount: ${amount}`,
+        );
+
+        const warpCore = contextFactory.getWarpCore();
+        const rebalancer = contextFactory.createRebalancer();
+        const originToken = warpCore.tokens.find((t) => t.chainName === origin);
```

Comment:
suggestion: considering that the rebalancer context is specific to a warp route, instead of having to manually filter over warp core, wouldn't it be cleaner to expose a method on the rebalancer context that allows to get the token info on a specific chain given the chain name/domain id?